modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import WithErrorHandler from './WithErrorHandler';
+import type { IDataTests } from './IDataTests';
+
+describe('WithErrorHandler', () => {
+  it('Should Render component', () => {
+    const successData = {
+      postTitle: 'a7la mesa 3lek!',
+    };
+    const mockComponent = jest.fn(function <T extends IDataTests.IDataSucccess>(
+      data: T,
+    ) {
+      return <div>{data.postTitle}</div>;
+    });
+    const Component = WithErrorHandler(mockComponent);
+    render(<Component data={successData} />);
+    expect(mockComponent).toBeCalled();
```

Comment:
I would assert that the final data is displayed, not the component has been called.